source/patterns/@aws-solutions-constructs/aws-fargate-sns/test/fargate-sns.test.ts

```diff
@@ -0,0 +1,362 @@
+    vpcProps: { cidr: '172.0.0.0/16' },
+    clusterProps: { clusterName },
+    containerDefinitionProps: { containerName },
+    fargateTaskDefinitionProps: { family: 'family-name' },
+    fargateServiceProps: { serviceName },
+    topicProps: { topicName },
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Service", {
+    LaunchType: 'FARGATE',
+    DesiredCount: 2,
+    DeploymentConfiguration: {
+      MaximumPercent: 150,
+      MinimumHealthyPercent: 75
+    },
+    PlatformVersion: ecs.FargatePlatformVersion.LATEST,
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Service", {
+    ServiceName: serviceName
+  });
+  expect(stack).toHaveResourceLike("AWS::ECS::TaskDefinition", {
+    Family: 'family-name'
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Cluster", {
+    ClusterName: clusterName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::SNS::Topic", {
+    TopicName: topicName
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::TaskDefinition", {
+    ContainerDefinitions: [
+      {
+        Essential: true,
+        Image: {
+          "Fn::Join": [
+            "",
+            [
+              "123456789012.dkr.ecr.us-east-1.",
+              {
+                Ref: "AWS::URLSuffix"
+              },
+              "/fake-repo:latest"
+            ]
+          ]
+        },
+        MemoryReservation: 512,
+        Name: containerName,
+        PortMappings: [
+          {
+            ContainerPort: 8080,
+            Protocol: "tcp"
+          }
+        ]
+      }
+    ]
+  });
+
+  expect(stack).toHaveResourceLike("AWS::EC2::VPC", {
+    CidrBlock: '172.0.0.0/16'
+  });
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::SNS::Topic', 1);
+  expect(stack).toCountResources('AWS::ECS::Service', 1);
+});
+
+test('New service/new topic, private API, new VPC', () => {
+
+  // An environment with region is required to enable logging on an ALB
+  const stack = new cdk.Stack(undefined, undefined, {
+    env: { account: "123456789012", region: 'us-east-1' },
+  });
+  const publicApi = false;
+
+  new FargateToSns(stack, 'test-construct', {
+    publicApi,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+    vpcProps: { cidr: '172.0.0.0/16' }
+  });
+
+  expect(stack).toHaveResourceLike("AWS::ECS::Service", {
+    LaunchType: 'FARGATE',
+    DesiredCount: 2,
+    DeploymentConfiguration: {
+      MaximumPercent: 150,
+      MinimumHealthyPercent: 75
+    },
+    PlatformVersion: ecs.FargatePlatformVersion.LATEST,
+  });
+
+  expect(stack).toHaveResourceLike("AWS::SNS::Topic", {
+
+  });
+
+  expect(stack).toHaveResourceLike("AWS::EC2::VPC", {
+    CidrBlock: '172.0.0.0/16'
+  });
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::SNS::Topic', 1);
+  expect(stack).toCountResources('AWS::ECS::Service', 1);
+});
+
+test('New service/existing topic, private API, existing VPC', () => {
```

Comment:
The only difference in this construct driven by the private API flag is the nature of the VPC if the construct creates the VPC (different types of subnets). This comment got me looking at this topic and I realized that the task always runs in a private or isolated subnet*, so the enableDns settings always need to be set to true regardless of publicApi. I've changed the code here and in aws-alb-fargate to reflect this.

* The service automatically chooses the most secure subnets available to run in (Isolated, Private,Public). The only way it will choose public is if the client passes in an existing VPC with only Public subnets.

Reply:
Also added checks to two tests to confirm that public and isolated VPCs are created correctly.